### PR TITLE
HELPS-3016 Rating component

### PR DESCRIPTION
**Pull request by BrettMugglinOlive**
### Changes made:

- Added the Rating component
- Added tests for it to the self test loop

### CODE REVIEW

File: ldk/javascript/examples/self-test-loop/src/config/whisper-group.ts [hunk 1]
```diff
@@ -371,4 +371,10 @@ export const whisperTestGroup = (): TestGroup =>
       10000,
       'Do the items have different widths than default?',
     ),
+    new LoopTest(
+      'Whisper Aptitude - Rating Test',
+      whisperTests.testRating,
+      10000,
+      'Tests for the Rating component',
+    ),
```
Comment (JosefBud-olive) [inline ldk/javascript/examples/self-test-loop/src/config/whisper-group.ts:379]: Same thing with alphabetizing whenever you get a sec

File: ldk/javascript/src/types/oliveHelps/whisper.d.ts [hunk 2]
```diff
@@ -350,6 +351,23 @@ declare namespace WhisperService {
...
+    icon?: string;
+    iconColor?: string;
+    max?: number;
+    name?: string;
+    onChange?: WhisperHandlerWithParam<number>;
+    precision?: number;
+    readOnly?: boolean;
+    size?: string;
+    value?: number;
+  };
```
Comment (JosefBud-olive) [inline ldk/javascript/src/types/oliveHelps/whisper.d.ts:369]: If all of these are optional, what happens if you use this component without any props?

**Review by JosefBud-olive — APPROVED** [review]

File: ldk/javascript/examples/self-test-loop/package.json [hunk 3]
```diff
@@ -164,4 +164,4 @@
     "ts-node": "^9.1.1",
     "xlsx": "^0.17.0"
   }
-}
```
Comment (iangallOlive) [inline ldk/javascript/examples/self-test-loop/package.json:167]: What were you thinking?!

**Review by iangallOlive — APPROVED** [review]

File: ldk/javascript/src/types/oliveHelps/whisper.d.ts (hunk 2 above)
Comment (BrettMugglinOlive) [reply ldk/javascript/src/types/oliveHelps/whisper.d.ts]: There are default values configured on the backend. It renders as a component with stars, 3 / 5 with the 3 being filled.

**Pull request merged**